test/helpers/search_qualifier_helper_test.rb

```diff
@@ -0,0 +1,124 @@
+require 'test_helper'
+
+class SearchQualifierHelperTest < ActionView::TestCase
+  test 'matches_id? should correctly match ids' do
+    ['123', '>=42', '<8'].each do |value|
+      assert matches_id?(value)
+    end
+
+    ['name', '3.14', '<=~', '-category'].each do |value|
+      assert_not matches_id?(value)
+    end
+  end
+
+  test 'matches_int? should correctly match integers' do
+    ['123', '>=42', '<8', '-404'].each do |value|
+      assert matches_int?(value)
+    end
+
+    ['name', '3.14', '<=~', '-tag'].each do |value|
+      assert_not matches_int?(value)
+    end
+  end
+
+  test 'matches_numeric? should correctly match numeric values' do
+    ['123', '>=42', '3.14'].each do |value|
+      assert matches_numeric?(value)
+    end
+
+    ['name', '-tag', '>-404'].each do |value|
+      assert_not matches_numeric?(value)
+    end
+  end
+
+  test 'matches_non_negative_int? should correctly match positive ints' do
+    ['123', '>=42', '9999', '0'].each do |value|
+      assert matches_non_negative_int?(value)
+    end
+
+    ['name', '-tag', '3.14'].each do |value|
+      assert_not matches_non_negative_int?(value)
+    end
+  end
+
+  test 'matches_status? should correctly match status' do
+    ['any', 'closed', 'open'].each do |value|
+      assert matches_status?(value)
+    end
+
+    ['name', '-category', '42'].each do |value|
+      assert_not matches_status?(value)
+    end
+  end
+
+  test 'numeric_value_sql should return correct operator and value' do
+    expected = {
+      '12345' => ['', '12345'],
+      '<12345' => ['<', '12345'],
+      '>=12345' => ['>=', '12345']
+    }
+    expected.each do |input, expect|
+      assert_equal expect, numeric_value_sql(input)
+    end
+  end
+
+  test 'date_value_sql should return correct operator, value, and timeframe' do
+    expected = {
+      '1' => ['', '1', 'MONTH'],
+      '1y' => ['', '1', 'YEAR'],
+      '<1y' => ['>', '1', 'YEAR'],
```

Comment:
`<1y` should produce `>`?  I hope this test failed?  (And similar for the next line.)

Reply:
This is correct, IIRC - a user-facing value of "<1y" ("less than one year ago") translates to an SQL query of "created_at > (one year ago)" - the operators switch.

Reply:
After taking a quick look, yeah, what @ArtOfCode- said - the test correctly checks that the operator is inverted

Reply:
Oh, I didn't realize it was just a relocation.  Thanks both of you for explaining.